src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -853,10 +897,78 @@ public SocketError ReceiveFrom(byte[] buffer, int offset, int count, ref SocketF
             }
             finally
             {
-                if (@event != null) @event.Dispose();
+                @event?.Dispose();
+            }
+        }
+
+        public unsafe SocketError ReceiveFrom(Span<byte> buffer, ref SocketFlags flags, byte[] socketAddress, ref int socketAddressLen, int timeout, out int bytesReceived)
```

Comment:
We can always go from byte[] to span, but we can't go back, and we can't store the span on the heap.  Per your question earlier, if we're ok using the BufferPtr variant and pinning the buffer longer than is potentially otherwise needed, then we could consolidate these paths, otherwise I don't see how we can without incurring additional cost.